### DISCUSSION TITLE
Error in compiling Moose Application

### DISCUSSION THREAD

**Discussion by Salma-Mao** [idaholab/moose#21934] [Q&A General] [answered]
Hello everyone,

   There are two problems.
   1. The core of my computer is 18  and he running memory is 64g. If the number of cores selected  exceeds 10, the calculation cannot be performed. I get the error.
   
   ===================================================================================
=   BAD TERMINATION OF ONE OF YOUR APPLICATION PROCESSES
=   PID 5191 RUNNING AT 940a54af5dff
=   EXIT CODE: 7
=   CLEANING UP REMAINING PROCESSES
=   YOU CAN IGNORE THE BELOW CLEANUP MESSAGES
===================================================================================
YOUR APPLICATION TERMINATED WITH THE EXIT STRING: Bus error (signal 7)
This typically refers to a problem with your application.
Please see the FAQ page for debugging suggestions
    
   2. The blocks of my case are 4. When I only output the mesh, it  displays 4 blocks. However,  the number of blocks in the output after the calculation  is 2.
    
![image](https://user-images.githubusercontent.com/54804946/186701338-44f33db8-62b2-478e-a168-ecc6da665af8.png)
![image](https://user-images.githubusercontent.com/54804946/186701781-1864a68e-9b6b-467a-bfbb-927caacbe20f.png)

**Comment by GiudGiud** [comment, marked as answer]
Hello

Can you try running with a distributed mesh? 

That's weird. Are you running something that deletes elements?

Guillaume

**Reply by Salma-Mao** [reply]
[https://mooseframework.inl.gov/source/userobject/CoupledVarThresholdElementSubdomainModifier.html](url) is used. I donot know how to run with a distributed mesh. Could you give me an example？

**Reply by GiudGiud** [reply]
if all the objects in your simulation are good for distributed meshes, just add `--distributed-mesh` on the command line

see this docs
[Replicated and Distributed Mesh](https://mooseframework.inl.gov/syntax/Mesh/#7d9b050e-7a34-4d05-9e93-f9d5a66aebac)

**Reply by Salma-Mao** [reply]
Do you know why  the number of cores selected exceeds 10, the calculation cannot be performed?  The cores of  my computer are 18.

**Reply by GiudGiud** [reply]
it could be that you are running out of memory. That s why we should try the distributed mesh.
You could add a postprocessor to report on memory usage

**Comment by Salma-Mao** [comment]
I only used 30% of the running memory. That's so  weird.

**Reply by GiudGiud** [reply]
the memory can spike at times. You d want to run that postprocessor that captures all spikes
https://mooseframework.inl.gov/source/postprocessors/MemoryUsage.html
